doc/source/structures/collections/range.rst

```diff
@@ -0,0 +1,91 @@
+.. _range:
+
+Range
+=====
+
+:struct:`Range` is a type that represents a sequence of scalar whole numbers. The sequence can start and finish at any whole number, can be either descending or ascending and can skip numbers.
+
+.. note::
+
+    This is one of the few places in kOS where there is a distinction between
+    decimal (floating point, or fractional) scalar numbers and whole (integer,
+    or round) scalar numbers.  Using a decimal scalar will not throw an error,
+    however it may give unexpected results due to rounding.
+
+There are 3 ways of constructing a :struct:`Range`:
+
+- :code:`RANGE(FROM, TO, STEP)`
+  Will create a sequence of numbers that starts with `FROM` and contains all numbers `N + STEP` such that `N` belongs to the sequence and that are smaller than `TO`. For example
+  :code:`RANGE(3,8,2)` will contain numbers 3, 5 and 7. If `FROM` > `TO` then the sequence will be descending. For example :code:`RANGE(2,-9,3)` will contain numbers 2, -1, -4 and -7.
```

Comment:
I'm not sure how I feel about this.  I didn't look at tomek's range feature in detail until now and I'm not sure I like it not including the endpoint and yet using the terminology "from...to".

"Count from 1 to 10" doesn't mean to stop at 9.  Even when a mathematical concept uses [0..n-1] rather than [1..n], the way you usually speak that aloud is "zero to N minus one", not "zero to N".

I get the mathematical reason to not include the endpoint, but maybe the terminology should be a bit different somehow.  Calling it "TO" really gives the wrong message.  You're not counting TO this value, but counting to just shy of it.  Maybe call them "begin" and "end"?